nomad/job_endpoint.go

```diff
@@ -1298,19 +1299,66 @@ func (j *Job) GetJobVersions(args *structs.JobVersionsRequest,
 			// Setup the output
 			reply.Versions = out
 			if len(out) != 0 {
+
+				var compareVersionNumber uint64
+				var compareVersion *structs.Job
+				var compareStatic bool
+
+				if args.DiffTagName != "" {
+					compareStatic = true
+					tagFound := false
+					for _, version := range out {
+						if version.TaggedVersion != nil && version.TaggedVersion.Name == args.DiffTagName {
+							tagFound = true
+							compareVersionNumber = version.Version
+							break
+						}
+					}
+					if !tagFound {
+						return fmt.Errorf("tag %q not found", args.DiffTagName)
+					}
+				} else if args.DiffVersion != "" {
+					compareStatic = true
+					compareVersionNumber, err = strconv.ParseUint(args.DiffVersion, 10, 64)
+					if err != nil {
+						return fmt.Errorf("failed to parse diff version: %v", err)
+					}
+				}
+
 				// Note: a previous assumption here was that the 0th job was the latest, and that we don't modify "old" versions.
 				// Adding version tags breaks this assumption (you can tag an old version, which should unblock /versions queries) so we now look for the highest ModifyIndex.
 				var maxModifyIndex uint64
 				for _, job := range out {
 					if job.ModifyIndex > maxModifyIndex {
 						maxModifyIndex = job.ModifyIndex
 					}
+					if compareStatic && job.Version == compareVersionNumber {
+						compareVersion = job
+					}
 				}
 				reply.Index = maxModifyIndex
+
+				if compareStatic && compareVersion == nil {
+					if args.DiffTagName != "" {
+						return fmt.Errorf("tag %q not found", args.DiffTagName)
+					}
+					return fmt.Errorf("version %q not found", args.DiffVersion)
+				}
```

Comment:
With this block it's not really clear to me what this `compareStatic` variable is for. It looks like you're setting this only when `DiffTagName` or `DiffVersion` are set, but not when `Diffs == true`, but won't these cases have been detected earlier on lines 1307-1326?

Reply:
Yeah you're exactly right, good catch!

Reply:
OH WAIT I just realized `compareStatic` is there because `compareVersion` is a uint64, not a pointer to a uint64, and therefore will instantiate at 0 here!

Reply:
Ok, that makes sense but I still don't understand where the word "static" comes from in the variable name.

Reply:
changed this to `compareSpecificVersion` instead